### ISSUE TITLE
BERT is not learning

### ISSUE THREAD

**Issue by Hepta-Col** [uds-lsv/BERT-LNL#2]
Hi, I am using your framework to train BERT under noisy labels. To see baseline performance, I am training BertWN on the IMDB dataset. I found that the accuracies don't change even a little bit. What happened? Could you help me?
![image](https://user-images.githubusercontent.com/62273225/206150695-e294eb7f-4889-44dd-a76c-23926ef3160b.png)
Here is how I print them: (in bert_wn_trainer.py)
![image](https://user-images.githubusercontent.com/62273225/206150919-8ebc262f-0436-487d-9270-ca9ad7348c49.png)
Here are the hyperparameters:
![image](https://user-images.githubusercontent.com/62273225/206151235-3d3ef8e4-1cf1-4051-acf2-5164555beb47.png)

**Comment by davidpig** [comment]
Hi,

thanks for reaching out.

I tried your configurations, the model works as expected at my side (screen shot attached). 

from the images you shared I see that both your (clean) validation and test accuracy are 1.0, could that be the case that you had some issues with your data (or it's just some dummy data?). I can imagine one case in which your model somehow remembers everything and stops updating, could that be the case?






![ss](https://user-images.githubusercontent.com/7349083/206703329-6e987867-66d4-43e9-950b-1789a6d3f598.png)

**Comment by Hepta-Col** [comment]
Thank you very much for answering my question! I am unsure whether I correctly prepared the data according to your instructions. Let's just focus on IMDB now, and I will show you the details.

First of all, I downloaded the IMDB dataset from this link: https://ai.stanford.edu/~amaas/data/sentiment/ 

Next, I created an "IMDB" folder and put the data I downloaded under an "src" folder under the "IMDB" folder, as shown below:
![image](https://user-images.githubusercontent.com/62273225/206706001-abe23019-6ae7-4a25-a381-85c6eebfec9e.png)

And also, I created a "txt_data" folder under "IMDB" folder as well:
![image](https://user-images.githubusercontent.com/62273225/206706128-3cd6df30-28c1-42cb-89ac-74bc8d218f3c.png)

Next, I should generate the files you required. Here is how I generate those files:
```
import os
import pickle as pkl
from tqdm import tqdm
import random
import numpy as np

root = "."

def create_imdb():
    target_dataset = 'IMDB'

    #! create test data
    test_data_file = os.path.join(root, target_dataset, 'txt_data', 'test.txt')
    test_label_file = os.path.join(root, target_dataset, 'txt_data', 'test_labels.pickle')
    
    with open(test_data_file, 'w') as test_data_f, \
            open(test_label_file, 'wb') as test_label_f:
        paths = []
        labels = []
        for dir in ['neg', 'pos']:
            dir_path = os.path.join(root, target_dataset, 'src', 'test', dir)
            for file_name in os.listdir(dir_path):  
                doc_path = os.path.abspath(os.path.join(dir_path, file_name))
                paths.append(doc_path)
                doc_label = 1 if dir == 'pos' else 0
                labels.append(doc_label)

        assert len(paths) == len(labels) == 25000
        labels = np.array(labels)
        
        selected_ids = random.sample(list(range(25000)), 2500)
        print("==> generating test data")
        for i in tqdm(selected_ids):
            test_data_f.write(paths[i] + '\n')
        pkl.dump(labels[selected_ids].tolist(), test_label_f)

    #! create train & validation data
    train_data_file = os.path.join(root, target_dataset, 'txt_data', 'train.txt')
    train_label_file = os.path.join(root, target_dataset, 'txt_data', 'train_labels.pickle')
    validation_data_file = os.path.join(root, target_dataset, 'txt_data', 'validation.txt')
    validation_label_file = os.path.join(root, target_dataset, 'txt_data', 'validation_labels.pickle')
    
    with open(train_data_file, 'w') as train_data_f, \
            open(train_label_file, 'wb') as train_label_f, \
                open(validation_data_file, 'w') as valid_data_f, \
                    open(validation_label_file, 'wb') as valid_label_f:
        paths = []
        labels = []
        for dir in ['neg', 'pos']:
            dir_path = os.path.join(root, target_dataset, 'src', 'train', dir)
            for file_name in os.listdir(dir_path):  
                doc_path = os.path.abspath(os.path.join(dir_path, file_name))
                paths.append(doc_path)
                doc_label = 1 if dir == 'pos' else 0
                labels.append(doc_label)

        assert len(paths) == len(labels) == 25000
        labels = np.array(labels)
        
        train_ids = random.sample(list(range(25000)), 21246)
        valid_ids = [i for i in list(range(25000)) if i not in train_ids]
        
        print("==> generating train data")
        for i in tqdm(train_ids):
            train_data_f.write(paths[i] + '\n')
        pkl.dump(labels[train_ids].tolist(), train_label_f)
        
        print("==> generating validation data")
        for i in tqdm(valid_ids):
            valid_data_f.write(paths[i] + '\n')
        pkl.dump(labels[valid_ids].tolist(), valid_label_f)
                    

if __name__ == '__main__':
    create_imdb()   # done
    print("DONE!")
```
After running this code, I obtained the following files:
![image](https://user-images.githubusercontent.com/62273225/206707584-dce3bf93-b5c8-4b76-8346-d38d4b4f3c56.png)

Did I misunderstand any of your requirements? Did you notice any inconsistency between my data preparation and yours? Thank you very much!

**Comment by davidpig** [comment]
Hi,
I quickly scanned through your code. I think your data processing code, as well as your data paths, are correct.

Please find the IMDB data I used for my experiments [here](https://drive.google.com/drive/folders/1eG4KZeE9efm4hzi7dZhQHiibUvHHokee?usp=share_link)

please let me know if the training still gets stuck.

**Comment by Hepta-Col** [comment]
Hi, thank you for your shared data! The experiments are going on as expected. Could you please also share the processed files for AGNews, Hausa, and Yoruba? I am not sure how you split these sets into train, validation, and test, but I hope to reproduce your results as close as possible. Thank you very much! I really appreciate your help

**Comment by davidpig** [comment]
Hi, 
sure, please find them [here](https://drive.google.com/drive/folders/1eG4KZeE9efm4hzi7dZhQHiibUvHHokee) 😊

**Comment by Hepta-Col** [comment]
Thank you! Could you share the hyperparameters configuration files for training each model on each dataset? I think at least the "num_training_steps" is different, right?

**Comment by davidpig** [comment]
Hi, sorry for the late reply.

For our vanilla models, we used the same hyperparameter settings on all datasets. The configurations are described in Section "Fine-tuning on Text Classification Task" in the appendix.

"num_training_steps" does not matter that much, because we, either way, use early stopping. "num_training_steps"  just gives us an upper bound of the training steps. You can set it to a large value if you also apply early stopping.

**Comment by Hepta-Col** [comment]
Hi! Thank you for getting back to me. I found that in your shared Google Drive, the contents in "AG-News" are the same as "IMDB", so there might be a mistake. Could you please re-upload the "AG-News" data? Thank you very much!

**Comment by davidpig** [comment]
Hi, sorry, you are right.

AG-News is reuploaded.  You can find the updated version using the same link.

**Comment by hhf-hd** [comment]
> Hi, sorry, you are right.
> 
> AG-News is reuploaded. You can find the updated version using the same link.

hello, could you reuploaded the ag news and imdb dataset  , the dataset in google drive is expired.
much appreciation ！

**Comment by davidpig** [comment]
Hi, sure. 

Please Please find the datasets [here](https://drive.google.com/file/d/14zNfLQ4zHTx9n66-iAMmS-Ew518sp53r/view?usp=sharing)

Also, I uploaded these datasets to Huggingface as well, please see, e.g., https://huggingface.co/datasets/davidpig/lnl-hausa

**Comment by hhf-hd** [comment]
> Hi, sure.
> 
> Please Please find the datasets [here](https://drive.google.com/file/d/14zNfLQ4zHTx9n66-iAMmS-Ew518sp53r/view?usp=sharing)
> 
> Also, I uploaded these datasets to Huggingface as well, please see, e.g., https://huggingface.co/datasets/davidpig/lnl-hausa

thank you very much !
